compat/inet_ntop.cc

```diff
@@ -150,37 +150,33 @@ inet_ntop4(const u_char *src, char *dst, size_t size)
 /* const char *
  * inet_ntop6(src, dst, size)
  *  convert IPv6 binary address into presentation (printable) format
- * author:
+ * original author:
  *  Paul Vixie, 1996.
  */
 static const char *
 inet_ntop6(const u_char *src, char *dst, size_t size)
 {
     /*
      * Note that int32_t and int16_t need only be "at least" large enough
-     * to contain a value of the specified size.  On some systems, like
-     * Crays, there is no such thing as an integer variable with 16 bits.
+     * to contain a value of the specified size.  On some systems
+     * there is no such thing as an integer variable with 16 bits.
      * Keep this in mind if you think this function should have been coded
      * to use pointer overlays.  All the world's not a VAX.
      */
-    char tmp[sizeof "ffff:ffff:ffff:ffff:ffff:ffff:255.255.255.255"], *tp;
-    struct { int base, len; } best, cur;
-    u_int words[NS_IN6ADDRSZ / NS_INT16SZ];
-    int i;
+    struct  {
+        int base = -1;
+        int len = 0;
+    } best, cur;
+    u_int words[NS_IN6ADDRSZ / NS_INT16SZ] = {};
```

Comment:
Since you have scoped this as better definitions. This should now be:
```suggestion
    unsigned int words[NS_IN6ADDRSZ / NS_INT16SZ] = {};
```